File: task-03/Elixir/prime_numbers_upto_n.exs
IO.puts("Enter a number: ")
n = String.trim(IO.gets("") || "") |> String.to_integer()

for num <- 2..n do
  is_prime = num in [2, 3] or not Enum.any?(2..trunc(:math.sqrt(num)), fn x -> rem(num, x) == 0 end)
  if is_prime, do: IO.puts("#{num} is prime")
end

File: task-03/Java/prime_numbers_upto_n.java
import java.util.Scanner;

public class prime_numbers_upto_n {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter the number n: ");
        int n = scanner.nextInt();

        System.out.println("Prime numbers up to " + n + " are: ");
        for (int i = 2; i <= n; i++) {
            boolean isPrime = true;
            for (int j = 2; j <= Math.sqrt(i); j++) {
                if (i % j == 0) {
                    isPrime = false;
                    break;
                }
            }
            if (isPrime) {
                System.out.print(i + " ");
            }
        }
    }
}